### ISSUE TITLE
mono/jit/jit.h: No such file or directory

### ISSUE THREAD

**Issue by tomerdomshlak** [mono/Embeddinator-4000#517]
Hi,

After installing the latest version with nuget,
I'm trying to compile the generated code with the following command:

./Embeddinator-4000.exe --gen=Java --platform=Android -c *filePath*

it results in the following error:

```
*root*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such file or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *root*\Embeddinator-4000.0.3.0\tools\c-support.h:30:0,
                 from *root*\Embeddinator-4000.0.3.0\tools\c-support.c:26:
*root*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such file or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *root*\Embeddinator-4000.0.3.0\tools\mono-support.c:26:0:
*root*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such fil
e or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *root*\Embeddinator-4000.0.3.0\tools\mono_embeddinator.h:32:0,
                 from *root*\Embeddinator-4000.0.3.0\tools\mono_embeddinator.c:29:
*root*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such fil
e or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.

Failed to compile generated code.


```

Any help please???
Thanks.

**Comment by tritao** [comment]
Is this on Windows or Mac?

**Comment by realvictorprm** [comment]
The question might be simple but effective .. Do you have Mono installed?

**Comment by tomerdomshlak** [comment]
It's on Windows.

Mono is installed, and the "missing" file exists under the path:

"C:\Program Files\Mono\include\mono-2.0\mono\jit\jit.h"

Looking at the source code, I see that it searches under the path "C:\Program Files\Mono".

**Comment by tritao** [comment]
Can you run with the verbose flag and paste here the entire output?

**Comment by tomerdomshlak** [comment]
`PS *hidden*\Embeddinator-4000.0.3.0\tools> ./Embeddinator-4000.exe --gen=Java --platform=Android -c -v "*hidden*.dll"
Parsing assemblies...
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
    Looking for Android SDK..
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\AndroidSdkDirectory not found.
      Key HKCU\SOFTWARE\Xamarin\MonoAndroid\PrivateAndroidSdkPath not found.
      Key HKCU\SOFTWARE\Android SDK Tools\Path not found.
      Key HKLM\SOFTWARE\Android SDK Tools\Path found:
    Path contains adb in \platform-tools (C:\Program Files (x86)\Android\android-sdk).
    Looking for Android NDK..
      Key HKCU\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\AndroidNdkDirectory not found.
      Key HKCU\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
      Key HKLM\SOFTWARE\Novell\Mono for Android\JavaSdkDirectory not found.
    Looking for Java 6 SDK..
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\CurrentVersion found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.6\JavaHome not found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.7\JavaHome not found.
      Key HKLM\SOFTWARE\JavaSoft\Java Development Kit\1.8\JavaHome found:
    Path contains jarsigner.exe in \bin (C:\Program Files (x86)\Java\jdk1.8.0_112).
    Found Android SDK. API levels: 19, 20, 21, 22, 23, 24
    Parsed 'Coronet.Xamarin.Common.dll'
Processing assemblies...
    Pass 'Embeddinator.CheckReservedKeywords'
    Pass 'Embeddinator.Passes.GenerateObjectTypesPass'
    Pass 'Embeddinator.Passes.GenerateArrayTypes'
    Pass 'CppSharp.Passes.CheckIgnoredDeclsPass'
... generating classes... 
    Pass 'Embeddinator.Passes.RenameDuplicatedDeclsPass'
... generating classes... 
    Pass 'CppSharp.Passes.CheckDuplicatedNamesPass'
    Pass 'Embeddinator.Passes.FieldToGetterSetterPropertyPass'
... generating classes... 
    Pass 'Embeddinator.Passes.FixMethodParametersPass'
    Pass 'Embeddinator.CheckReservedKeywords'
Generating binding code...
    Generated: *hidden*.Common.h
    Generated: *hidden*.Common.c
    Generated: c-support.c
    Generated: c-support.h
    Generated: embeddinator.h
    Generated: glib.c
    Generated: glib.h
    Generated: mono-support.c
    Generated: mono-support.h
    Generated: mono_embeddinator.c
    Generated: mono_embeddinator.h
Compiling binding code...
    Invoking: C:\ProgramData\Microsoft\AndroidNDK64\android-ndk-r13b\toolchains\arm-linux-androideabi-4.9\prebuilt\windo
ws-x86_64\bin\arm-linux-androideabi-gcc.exe --sysroot="C:\ProgramData\Microsoft\AndroidNDK64\android-ndk-r13b\platforms\
android-24\arch-arm" -fdiagnostics-color -DMONO_EMBEDDINATOR_DLL_EXPORT -I"C:\Program Files (x86)\Mono\include\mono-2.0"
 -L"*hidden*\Embeddinator-4000.0.3.0\external\Xamarin.Android\lib\xbuild\Xamarin\Android\lib\armeabi"
-lmonosgen-2.0 -lmono-android.release *hidden*\Embeddinator-4000.0.3.0\tools\Coronet.Xamarin.Common.c
C:\Workspace\coronet\build\Embeddinator-4000.0.3.0\tools\c-support.c C:\Workspace\coronet\build\Embeddinator-4000.0.3.0\
tools\glib.c *hidden*\Embeddinator-4000.0.3.0\tools\mono-support.c *hidden*d\Embeddin
ator-4000.0.3.0\tools\mono_embeddinator.c --std=c99 -shared -o *hidden*\Embeddinator-4000.0.3.0\tools\
android\jni\armeabi\libCoronet.Xamarin.Common.so
        In file included from *hidden*\Embeddinator-4000.0.3.0\tools\mono_embeddinator.h:32:0,
                 from *hidden*\Embeddinator-4000.0.3.0\tools\Coronet.Xamarin.Common.h:10,
                 from *hidden*\Embeddinator-4000.0.3.0\tools\Coronet.Xamarin.Common.c:7:
*hidden*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such fil
e or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *hidden*\Embeddinator-4000.0.3.0\tools\c-support.h:30:0,
                 from *hidden*\Embeddinator-4000.0.3.0\tools\c-support.c:26:
*hidden*\build\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such file or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *hidden*\build\Embeddinator-4000.0.3.0\tools\mono-support.c:26:0:
*hidden*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such file or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.
In file included from *hidden*\Embeddinator-4000.0.3.0\tools\mono_embeddinator.h:32:0,
                 from *hidden*\Embeddinator-4000.0.3.0\tools\mono_embeddinator.c:29:
*hidden*\Embeddinator-4000.0.3.0\tools\mono-support.h:187:26: fatal error: mono/jit/jit.h: No such fil
e or directory
 #include <mono/jit/jit.h>
                          ^
compilation terminated.

Failed to compile generated code.`

**Comment by tritao** [comment]
Thanks, I believe the fix I linked above should deal with this, so please give it a try with master.

If it still fails, then feel free to re-open and I'll take a further look.
